### PR TITLE
Version 1.2.0: Big refactor

### PR DESCRIPTION
**Pull request by cesaraugustogarcia**
Version 1.2.0: Big refactor

### CODE REVIEW

**Review by santjc — APPROVED** [review]
Awesome work Cesar!

**Comment by cesaraugustogarcia** [comment]
> Awesome work Cesar! The next feature could be to improve error handling in each function, so we could have more detailed logs in case of a misfunction.

Totally agree!

File: functions/getPropertiesFromArguments.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+];
+const OPTIONAL_ARGS = ["testURL"];
+
+/**
+ * Read activity arguments in the shape of --param=value from script invocation
+ */
+const getPropertiesFromArguments = (arguments) => {
+  if (typeof arguments === "object" && !Array.isArray(arguments)) {
+    return arguments;
+  } else {
```
Comment (juansebastiancodes) [inline functions/getPropertiesFromArguments.js:31]: Maybe this is an unncesary else

**Review by juansebastiancodes — APPROVED** [review]

File: functions/getPropertiesFromArguments.js (hunk 1 above)
Comment (cesaraugustogarcia) [reply functions/getPropertiesFromArguments.js]: Changed!

**Pull request merged**